pkg/input/types/http_test.go

```diff
@@ -65,3 +66,23 @@ func TestParseHttpRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestUnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name       string
+		rawJSONStr string
+	}{
+		{"basic url", `{"url": "example.com"}`},
+		{"basic url with scheme", `{"url": "http://example.com"}`},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var rr RequestResponse
+			err := json.Unmarshal([]byte(tc.rawJSONStr), &rr)
```

Comment:
Can we test `func (rr *RequestResponse) UnmarshalJSON(data []byte) error {}` instead?